File: android-compiler/src/main/java/com/pranav/lib_android/code/disassembler/JavapDisassembler.java
package com.pranav.lib_android.code.disassembler;

import com.sun.tools.javap.JavapTask;

import java.io.StringWriter;
import java.util.ArrayList;

public class JavapDisassembler {

    final String path;

    public JavapDisassembler(String classFile) {
        path = classFile;
    }

    public String disassemble() throws Throwable {
        // Create an arraylist for storing javap arguments
        var args = new ArrayList<String>();
        args.add("-c");
        args.add(path);
        // Create a StringWriter object that will store the output
        var writer = new StringWriter();
        // Create a JavapTask to handle the arguments
        var task = new JavapTask();
        task.handleOptions(args.toArray(new String[0]));
        task.setLog(writer);
        task.run();
        // return the disassembled file as string
        return writer.toString();
    }
}

File: android-compiler/src/main/java/com/pranav/android/code/disassembler/JavapDisassembler.kt
package com.pranav.android.code.disassembler

import com.sun.tools.javap.JavapTask

import java.io.StringWriter
import java.util.ArrayList

class JavapDisassembler(filePath: String) {

    private val path: String

    init {
        path = filePath
    }

    @Throws(Throwable::class)
    fun disassemble() : String {
        // Create an arraylist for storing javap arguments
        val args = listOf(
                "-c",
                path
        )
        // Create a StringWriter object that will store the output
        val writer = StringWriter()
        // Create a JavapTask to handle the arguments
        val task = JavapTask()
        task.handleOptions(args.toTypedArray())
        task.setLog(writer)
        task.run()
        // return the disassembled file as string
        return writer.toString()
    }
}
